ocs_ci/deployment/helpers/lso_helpers.py

```diff
@@ -42,52 +42,7 @@ def setup_local_storage(storageclass):
     ocp_version = version.get_semantic_ocp_version_from_config()
     ocs_version = version.get_semantic_ocs_version_from_config()
     ocp_ga_version = get_ocp_ga_version(ocp_version)
-    if not ocp_ga_version:
-        optional_operators_data = list(
-            templating.load_yaml(
-                constants.LOCAL_STORAGE_OPTIONAL_OPERATORS, multi_document=True
-            )
-        )
-        optional_operators_yaml = tempfile.NamedTemporaryFile(
-            mode="w+", prefix="optional_operators", delete=False
-        )
-        if config.DEPLOYMENT.get("optional_operators_image"):
-            for _dict in optional_operators_data:
-                if _dict.get("kind").lower() == "catalogsource":
-                    _dict["spec"]["image"] = config.DEPLOYMENT.get(
-                        "optional_operators_image"
-                    )
-        if config.DEPLOYMENT.get("disconnected"):
-            # in case of disconnected environment, we have to mirror all the
-            # optional_operators images
-            icsp = None
-            for _dict in optional_operators_data:
-                if _dict.get("kind").lower() == "catalogsource":
-                    index_image = _dict["spec"]["image"]
-                if _dict.get("kind").lower() == "imagecontentsourcepolicy":
-                    icsp = _dict
-            mirrored_index_image = (
-                f"{config.DEPLOYMENT['mirror_registry']}/"
-                f"{index_image.split('/', 1)[-1]}"
-            )
-            prune_and_mirror_index_image(
-                index_image,
-                mirrored_index_image,
-                constants.DISCON_CL_REQUIRED_PACKAGES,
-                icsp,
-            )
-            _dict["spec"]["image"] = mirrored_index_image
-        templating.dump_data_to_temp_yaml(
-            optional_operators_data, optional_operators_yaml.name
-        )
-        with open(optional_operators_yaml.name, "r") as f:
-            logger.info(f.read())
-        logger.info(
-            "Creating optional operators CatalogSource and ImageContentSourcePolicy"
-        )
-        run_cmd(f"oc create -f {optional_operators_yaml.name}")
-        logger.info("Sleeping for 60 sec to start update machineconfigpool status")
-        wait_for_machineconfigpool_status("all")
+    create_optional_operators_catalogsource_non_ga()
```

Comment:
Shouldn't we call this function only for non ocp_ga_versions. 
```
if not ocp_ga_version:
  create_optional_operators_catalogsource_non_ga()
```

Reply:
@OdedViner moved this check into the function, so that it won't do anything when run on GA version. I would not do that myself, but at the same time I decided to not object it in the review (as I wouldn't approve anything otherwise).

Reply:
Yes, I did see that Oded is handling it in the function itself. 
If we call setup_local_storage(), below two lines of code are executed twice, one in the setup_local_storage and again in create_optional_operators_catalogsource_non_ga.
```
ocp_version = version.get_semantic_ocp_version_from_config()
ocp_ga_version = get_ocp_ga_version(ocp_version)
```
In the setup_local_storage() itself we will come to know whether the ocp version is a GA one or non GA. So if the ocp version is GA version we can right away skip entering create_optional_operators_catalogsource_non_ga().

I felt good to have this handled , if you all think that is fine, feel free to ignore this comment.

Reply:
I agree with @PrasadDesala

Reply:
done